### PR TITLE
docs,esp8266,esp32: Replace deprecated network interface constants.

### PR DESCRIPTION
**Pull request by projectgus**
### Summary

As noted in https://github.com/micropython/micropython/pull/16119#issuecomment-2453639664 the deprecated `network.STA_IF` and `network.AP_IF` constants were still listed in a number of places instead of the recommended `network.WLAN.IF_STA` and `network.WLAN_IF_AP` constants.

This PR updates all remaining instances.

This PR has a sibling in micropython-lib: https://github.com/micropython/micropython-lib/pull/933

### Testing

* I haven't tested these changes on device, but I believe the new constants exist on all ports with Wi-Fi and these changes were 95% done by regex so there should be minimal chance of breakage.

### CODE REVIEW

File: docs/esp8266/tutorial/network_basics.rst [hunk 1]
```diff
@@ -57,8 +57,8 @@ connect to your WiFi network::
 
     def do_connect():
         import network
-        sta_if = network.WLAN(network.STA_IF)
-        if not sta_if.isconnected():
+        sta_if = network.WLAN(network.WLAN.IF_STA)
+         if not sta_if.isconnected():
```
Comment (dpgeorge) [inline docs/esp8266/tutorial/network_basics.rst:61]: stray space added??

File: docs/esp8266/tutorial/network_basics.rst [hunk 2]
```diff
@@ -1,14 +1,14 @@
 Network basics
 ==============
 
-The network module is used to configure the WiFi connection.  There are two WiFi
+The :class:`network.WLAN` class in the :mod:`network` module is used to configure the WiFi connection.  There are two WiFi
```
Comment (dpgeorge) [inline docs/esp8266/tutorial/network_basics.rst:4]: maybe line break between "module" and "is used" to keep the lines somewhat the same length in this paragraph without needing to reflow the whole thing?

**Review by dpgeorge — APPROVED** [review]
Looks good, thanks!

There are a few commit messages with two signed-off lines that you may want to fix.

File: docs/esp8266/tutorial/network_basics.rst (hunk 2 above)
Comment (projectgus) [reply docs/esp8266/tutorial/network_basics.rst]: I ended up reflowing this paragraph and adjusting a couple of others for long lines

**Review by dpgeorge — APPROVED** [review]

**Pull request merged**